enterprise/backend/src/metabase_enterprise/transforms/jobs.clj

```diff
@@ -25,27 +26,50 @@
                (into more-transforms (get ordering current-transform))))
       found)))
 
+(defn- first-transform [[initial & more]]
+  (when initial
+    (reduce (fn [best current]
+              (if (neg? (compare (:created_at current) (:created_at best)))
+                current
+                best))
+            initial
+            more)))
+
+(defn- next-transform [ordering transforms-by-id complete]
+  (->> (transforms.ordering/available-transforms ordering #{} complete)
+       (map transforms-by-id)
+       first-transform))
```

Comment:
if `transforms.ordering/available-transforms` returned the ids in order of increasing transform `created_at` values, then we could just pick the first. That would work nicely with the lazy sequence returned by `transforms.ordering/available-transforms`.